test/e2e/istio-csr_test.go

```diff
@@ -0,0 +1,300 @@
+										"-cacert", "/etc/certs/root-cert.pem",
+										fmt.Sprintf("cert-manager-istio-csr.%s.svc:443", ns.Name),
+										"istio.v1.auth.IstioCertificateService/CreateCertificate",
+									},
+									VolumeMounts: []corev1.VolumeMount{
+										{Name: "cert-volume", MountPath: "/etc/certs"},
+										{Name: "proto", MountPath: "/proto"},
+									},
+								},
+							},
+							Volumes: []corev1.Volume{
+								{
+									Name: "cert-volume",
+									VolumeSource: corev1.VolumeSource{
+										ConfigMap: &corev1.ConfigMapVolumeSource{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: "istio-ca-root-cert",
+											},
+										},
+									},
+								},
+								{
+									Name: "proto",
+									VolumeSource: corev1.VolumeSource{
+										ConfigMap: &corev1.ConfigMapVolumeSource{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: "proto-cm",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			_, err = clientset.BatchV1().Jobs(ns.Name).Create(context.TODO(), job, metav1.CreateOptions{})
+			Expect(err).Should(BeNil())
+			defer clientset.BatchV1().Jobs(ns.Name).Delete(ctx, job.Name, metav1.DeleteOptions{})
+
+			By("waiting for the job to be completed")
+			err = pollTillCompleted(ctx, clientset, ns.Name, fmt.Sprintf("app=%s", grpcAppName))
+			Expect(err).Should(BeNil())
+
+			By("fetching logs of the grpcurl job")
+			pods, err := clientset.CoreV1().Pods(ns.Name).List(context.TODO(), metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("app=%s", grpcAppName),
+			})
+			Expect(err).Should(BeNil())
+
+			By("reading logs of the grpcurl job")
+			for _, pod := range pods.Items {
+				req := clientset.CoreV1().Pods(ns.Name).GetLogs(pod.Name, &corev1.PodLogOptions{})
+				logs, err := req.Stream(context.TODO())
+				Expect(err).Should(BeNil())
+
+				defer logs.Close()
+
+				logData, err := io.ReadAll(logs)
+				Expect(err).Should(BeNil())
+
+				var entry LogEntry
+				err = json.Unmarshal(logData, &entry)
+				Expect(err).Should(BeNil())
+
+				By("validating each certificate")
+				for _, certPEM := range entry.CertChain {
+					block, _ := pem.Decode([]byte(certPEM))
+					cert, err := x509.ParseCertificate(block.Bytes)
+					Expect(err).Should(BeNil())
+
+					Expect(cert.Issuer.CommonName).Should(Equal("my-selfsigned-ca"))
+
+					// Validate the certificate's validity period
+					By("validating the certificate validity period")
+					now := time.Now()
+					Expect(cert.NotBefore).Should(BeTemporally("<=", now), "Certificate's NotBefore should be in the past or present")
+					Expect(cert.NotAfter).Should(BeTemporally(">", now), "Certificate's NotAfter should be in the future")
+
+				}
+			}
+		})
+	})
+})
+
+func pollTillCompleted(ctx context.Context, clientset *kubernetes.Clientset, namespace, labelSelector string) error {
```

Comment:
no need to list the pods and check their status,
just poll the Job object itself and look for it's status to see completion status.